playbooks/roles/caasp4/tasks/_assert_packages.yml

```diff
@@ -0,0 +1,73 @@
+---
+#
+# (c) Copyright 2019 SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
```

Comment:
you can assert here that its running under sles15 with a nice fail message, so we dont need so many `when` in the following tasks.

Reply:
I was thinking common-vars is the place where version specific entries (logic) need to reside. [https://github.com/SUSE-Cloud/socok8s/pull/711/files#diff-0d7e3c79df5eb0ccb93627cfaa14b956R50]

If there is host with sles15-sp2 version, then we just add entry in common-vars. No need to check specific version here. 
And if needed package repo entry is not added, so user need to go manual route and then it will still pass here as we are not checking for specific version

Reply:
This way allows user to NOT have SLE 15-sp1 as runner host as we are not explicitly checking for specific version.
@evrardjp  ^^^